### PR TITLE
Avoid traceback with AssertionError: Failed to update row (bsc#1172558)

### PR DESCRIPTION
**Pull request by moio**
## What does this PR change?

It prevents a traceback by removing a wrong assertion that failed in cases where no error really happened.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **bugfix**
- [x] **DONE**

## Test coverage
- Cucumber tests previously failing are now green (problem was reproduced locally running `secondary/srv_patches_page.feature` in isolation - errors would pop up most runs)

- [x] **DONE**

## Links

Fixes https://bugzilla.suse.com/show_bug.cgi?id=1172558
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by moio** [comment]
@chiaradiamarcelo: [this change you contributed some time ago](https://github.com/uyuni-project/uyuni/commit/bb406bc#diff-92018908bc70886767765065bfc1f240R41) ([downstream PR](https://github.com/SUSE/spacewalk/pull/5006)) adds `ON CONFLICT DO NOTHING` to some `INSERT` statements, which is the correct fix for the bug at hand.

Problem is, the assertion I am removing here would be violated in such cases, as the number of `INSERT`ed rows would be 0 and not 1. We can thus disregard the assert.

Am I missing anything?

File: java/code/src/com/redhat/rhn/domain/rhnset/RhnSetFactory.java [hunk 1]
```diff
@@ -184,22 +184,19 @@ private static void executeMode(RhnSetElement elem,
...
+            el2.executeUpdate(params);
         }
         else {
             params.put("el_three", elem.getElementThree());
             params.put("el_two", elem.getElementTwo());
-            count = el3.executeUpdate(params);
+            el3.executeUpdate(params);
         }
-
-        assert count == 1 : "Failed to update row";
```
Comment (srbarrios) [inline java/code/src/com/redhat/rhn/domain/rhnset/RhnSetFactory.java:202]: Fine to remove the assert, but... it does mean we are not checking the output of executeUpdate method?
What if the row failed updating it? :thinking:

Comment (moio) [reply java/code/src/com/redhat/rhn/domain/rhnset/RhnSetFactory.java:202]: We do not have much to check. The update is restricted to just one row so the only possible outputs are two: 1 (row updated) or 0 (rows updated) - there cannot be more.

Before commit bb406bc, an output of 0 would signal an error, as we were definitely expecting 1 row to be updated. After the commit, we changed the query and both 1 and 0 are OK as outputs.

So there is nothing really left to check here, and in particular, failing on 0 is wrong, thus I am removing the check.

**Comment by moio** [comment]
jUnit tests are failing but causes look unrelated. Merging now.

**Pull request merged**